server/src/uuid/model/page/mod.rs

```diff
@@ -117,6 +117,113 @@ impl UuidFetcher for Page {
     }
 }
 
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PageAddRevisionPayload {
+    pub content: String,
+    pub title: String,
+    pub page_id: i32,
+    pub user_id: i32,
+}
+
+#[derive(Error, Debug)]
+pub enum PageAddRevisionError {
+    #[error("Revision could not be added because of a database error: {inner:?}.")]
+    DatabaseError { inner: sqlx::Error },
+    #[error("Revision could not be added out because of an event error: {inner:?}.")]
+    CheckoutRevisionError {
+        inner: Box<PageCheckoutRevisionError>,
+    },
+    #[error("Revision could not be added because of an UUID error: {inner:?}.")]
+    UuidError { inner: UuidError },
+    #[error("Revision could not be added because page was not found.")]
+    PageNotFound,
+}
+impl From<sqlx::Error> for PageAddRevisionError {
+    fn from(inner: sqlx::Error) -> Self {
+        Self::DatabaseError { inner }
+    }
+}
+
+impl From<PageCheckoutRevisionError> for PageAddRevisionError {
+    fn from(inner: PageCheckoutRevisionError) -> Self {
+        Self::CheckoutRevisionError {
+            inner: Box::new(inner),
+        }
+    }
+}
+
+impl From<UuidError> for PageAddRevisionError {
+    fn from(error: UuidError) -> Self {
+        match error {
+            UuidError::DatabaseError { inner } => inner.into(),
+            inner => Self::UuidError { inner },
+        }
+    }
+}
+
+impl Page {
+    pub async fn add_revision<'a, E>(
+        payload: PageAddRevisionPayload,
+        executor: E,
+    ) -> Result<Uuid, PageAddRevisionError>
+    where
+        E: Executor<'a>,
+    {
+        let mut transaction = executor.begin().await?;
+
```

Comment:
Should I add any checks if content and title are empty?

Reply:
Good point. I'd at least do the validation in the API. Might also be a topic for the dev meeting on how we want to handle validation in general. IMHO:

- API is the consumer-facing service and should handle most of the user-input validation.
- Database layer may assume that API has already done the validation.

I'm open for other suggestions, though. As long as we stay mostly consistent with our approach, I'm fine with it.